### PR TITLE
Changes to Lightcurve saving + SVI modularity

### PR DESCRIPTION
**Pull request by kdesoto-astro**
## Change Description
Moves some of SVI to helper function for easy ELASTICC filter use. Also edits Lightcurve file saving and loading to preserve name and sn_class fields. Also some tiny miscellaneous plotting function changes for the paper I'm working on.

## Code Quality
- [x] I have read the Contribution Guide
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

### CODE REVIEW

File: src/superphot_plus/elasticc_class.py
Comment (delucchi-cmu) [inline src/superphot_plus/elasticc_class.py:1]: This looks identical to the existing `SupernovaClass`, but with some methods removed. Are there other changes you intend to make, e.g. to alternative namings?

File: src/superphot_plus/lightcurve.py [hunk 1]
```diff
@@ -322,19 +322,20 @@ def from_file(cls, filename, ref_band="r", t0_lim=None, shift_time=True):
         """
         if not os.path.exists(filename):
             raise FileNotFoundError(f"ERROR: File does not exist {filename}")
-        curve_name = filename.split("/")[-1][:-4]
 
         # Load the data as a numpy array.
-        npy_array = np.load(filename)
-        arr = npy_array["arr_0"]
+        npy_array = np.load(filename, allow_pickle=True)
```
Comment (delucchi-cmu) [inline src/superphot_plus/lightcurve.py:327]: You will also need to update the file in test data.

File: src/superphot_plus/elasticc_class.py
Comment (kdesoto-astro) [reply src/superphot_plus/elasticc_class.py:1]: Yes! Sorry that's just a placeholder for now. I'll be adding new mappings there

File: src/superphot_plus/lightcurve.py (hunk 1 above)
Comment (kdesoto-astro) [reply src/superphot_plus/lightcurve.py:327]: I did! But I don't know if it uploaded correctly

Comment (kdesoto-astro) [reply src/superphot_plus/lightcurve.py:327]: It didn't - whoops!

**Review by delucchi-cmu — APPROVED** [review]

**Pull request merged**